procurement_auto_create_group/models/stock_rule.py

```diff
@@ -16,15 +16,18 @@ def _onchange_group_propagation_option(self):
         if self.group_propagation_option != "propagate":
             self.auto_create_group = False
 
+    def _get_auto_procurement_group(self, product_id):
+        group_data = self._prepare_auto_procurement_group_data(product_id)
```

Comment:
If `product_id` is a recordset, I would name it `product` here (for the `_get_rule` method we cannot as it's a std method).